src/Psalm/Internal/LanguageServer/LanguageServer.php

```diff
@@ -400,7 +400,8 @@ public function initialize(
             $this->path_mapper->configureClientRoot($this->getPathPart($rootUri));
         }
 
-        return call(
+        /** @var Promise<InitializeResult> $promise */
+        $promise = call(
```

Comment:
This avoid
```
The type 'Amp\Promise<LanguageServerProtocol\InitializeResult>|Amp\Promise<mixed>' is more general than the declared return type 'Amp\Promise<LanguageServerProtocol\InitializeResult>' for Psalm\Internal\LanguageServer\LanguageServer::initialize
```

This is technically true because the conditional return type is
```
(T is Promise ? false : (T is \Generator ? (TGenerator is Promise ? Promise<TGeneratorPromise> : Promise<TGeneratorReturn>) : int))
```
and the check `InitializeResult is Promise` cannot be resolved since `InitializeResult&Promise` would be a possible type.